restapi/api/device_endpoints.go

```diff
@@ -64,3 +66,54 @@ func Screenshot(c *gin.Context) {
 	c.Header("Content-Type", "image/png")
 	c.Data(http.StatusOK, "application/octet-stream", b)
 }
+
+// Change the current device location
+// @Summary      Change the current device location
+// @Description Change the current device location to provided latitude and longtitude
+// @Tags         general_device_specific
+// @Produce      json
+// @Param        latitude  query      string  true  "Location latitude"
+// @Param        longtitude  query      string  true  "Location longtitude"
+// @Success      200  {object}  GenericResponse
+// @Failure		 422  {object}  GenericResponse
+// @Failure		 500  {object}  GenericResponse
+// @Router       /device/{udid}/setlocation [post]
+func SetLocation(c *gin.Context) {
+	device := c.MustGet(IOS_KEY).(ios.DeviceEntry)
+	latitude := c.Query("latitude")
+	if latitude == "" {
+		c.JSON(http.StatusUnprocessableEntity, GenericResponse{Error: "latitude query param is missing"})
+		return
+	}
+
+	longtitude := c.Query("longtitude")
+	if longtitude == "" {
+		c.JSON(http.StatusUnprocessableEntity, GenericResponse{Error: "longtitude query param is missing"})
+		return
+	}
+
+	err := simlocation.SetLocation(device, latitude, longtitude)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, GenericResponse{Error: err.Error()})
+	} else {
```

Comment:
I am not the biggest fan of else statements.
I usually prefer this pattern inspired by swift guards:
if err != nil{
...
return
}
rest of the code

Reply:
I also usually do this, but my other development efforts get in the way :D Updated it in both endpoints

Reply:
thanks!